From da2011bcb54ae8c94509dfd14ceb693e779c459d Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:12:49 -0400
Subject: [ethernet] i40evf: don't violate scope

Message-id: <1409663591-32531-127-git-send-email-sassmann@redhat.com>
Patchwork-id: 90602
O-Subject: [RHEL7.1 PATCH 126/148] i40evf: don't violate scope
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

Move a declaration up one level so we don't dereference it out of scope.
This didn't cause any panics, but the details->async field would
mysteriously disappear, causing unnecessary delays when sending AQ
commands. Also, the code is just plain wrong.

Change-ID: I753f64f13c55e5d75ea4351e29b14fb53b2f0104
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 4334edf53a032f54403d1f367b501502a8c61e93)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40e_common.c b/drivers/net/ethernet/intel/i40evf/i40e_common.c
index a43155a..4ea90bf 100644
--- a/drivers/net/ethernet/intel/i40evf/i40e_common.c
+++ b/drivers/net/ethernet/intel/i40evf/i40e_common.c
@@ -551,6 +551,7 @@ i40e_status i40e_aq_send_msg_to_pf(struct i40e_hw *hw,
     struct i40e_asq_cmd_details *cmd_details)
 {
  struct i40e_aq_desc desc;
+ struct i40e_asq_cmd_details details;
  i40e_status status;
 
  i40evf_fill_default_direct_cmd_desc(&desc, i40e_aqc_opc_send_msg_to_pf);
@@ -565,7 +566,6 @@ i40e_status i40e_aq_send_msg_to_pf(struct i40e_hw *hw,
   desc.datalen = cpu_to_le16(msglen);
  }
  if (!cmd_details) {
-  struct i40e_asq_cmd_details details;
   memset(&details, 0, sizeof(details));
   details.async = true;
   cmd_details = &details;
-- 
1.7.1